sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -42,6 +43,29 @@ namespace Azure { namespace Core { namespace Http {
     }
   } // namespace Details
 
+  /*********************  Exceptions  **********************/
+  /**
+   * @brief Couldn't resolve HTTP host.
+   */
+  struct CouldNotResolveHostException : public Azure::Core::RequestFailedException
```

Comment:
Unlike other exceptions we define, this one is a bit too specific to one particular error. Maybe we can generalize it a bit (and in the process come up with a simpler name).

Reply:
There is more benefit to have errors as particular as possible.
That's for easier debug and info for API callers.

Reply:
Having one exception for every single specific error is strange though and doesn't hit the right balance. These aren't error codes. And the user may want to use them in try/catch blocks. That's the whole point of an exception hierarchy, which is different from linear error codes.